#!/usr/bin/env python
PACKAGE = "hydrus"

from dynamic_reconfigure.parameter_generator_catkin import *
TORSION_NULL_SPACE_SHIFT_THREASH = 0
TORSION_NULL_SPACE_SHIFT_LIMIT_RATIO = 1
TORSION_NULL_SPACE_SHIFT_MIX_LIMIT = 2
TORSION_NULL_SPACE_SHIFT_POSITION_PENALTY = 3
TORSION_NULL_SPACE_SHIFT_ALPHA = 4

gen = ParameterGenerator()

gen.add("null_space_shift_thresh", double_t, TORSION_NULL_SPACE_SHIFT_THREASH, "LQI Torsion null space shift thresh", 0.0, 0, 1.0)
gen.add("null_space_shift_limit_ratio", double_t, TORSION_NULL_SPACE_SHIFT_LIMIT_RATIO, "LQI Torsion null space shift limit ratio", 0.0, 0, 1000.0)
gen.add("null_space_shift_mix_limit", double_t, TORSION_NULL_SPACE_SHIFT_MIX_LIMIT, "LQI Torsion null space shift mix limit", 0.0, 0, 1.0)
gen.add("nlopt_position_penalty", double_t, TORSION_NULL_SPACE_SHIFT_POSITION_PENALTY, "LQI Torsion null space shift position panalty", 0.0, 0, 1.0)
gen.add("nlopt_alpha", double_t, TORSION_NULL_SPACE_SHIFT_ALPHA, "LQI Torsion null space shift alpha", 0.0, 0, 10.0)

exit(gen.generate(PACKAGE, "hydrus", "torsionNullSpaceShifter"))
